### ISSUE TITLE
Plotting error with total cross section

### ISSUE THREAD

**Issue by Shimwell** [openmc-dev/openmc#1561]
Hi all,

While creating some plots using the ```openmc.plot_xs``` function it was noticed that the total cross sections are a little lower than other cross sections at some energies. [A google groups post ](https://groups.google.com/forum/?nomobile=true#!topic/openmc-users/7oea0UjsjCc)was created and @eepeterson also noticed the issue. 

After a little digging around the code a potential error was spotted in the endf.py SUM_RULES for rule number 1. The current value is  ```1: [2, 3]``` when perhaps the combination of cross sections should be 2, 4, 5, 11, 16-18, 22-26, 28-37, 41-42, 44-45, 102-117. This is defined in the [Sum rules for ENDF part 0.5.12](https://www.oecd-nea.org/dbdata/data/endf102.htm) .

I was thinking part of the solution could be to change [ this line](https://github.com/openmc-dev/openmc/blob/2f0a8941316bffb5c5a17ea525915ea5c2cb3ff9/openmc/data/endf.py#L50) in endf.py from

```SUM_RULES = {1: [2, 3],```

to something like this

```

SUM_RULES = {1: [2, 4, 5, 11, 16,17, 18, 22, 23, 24, 25, 26,
                 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 41, 
                 42, 44, 45, 102, 103, 104, 105, 106, 107,
                 108, 109, 110, 111, 112, 113, 114, 115, 
                 116, 117],
```

However this appears to have knock on consequences as 'total' appears to be used for total neutron   production from fission and is summed from the prompt and delayed on line [456 of plotter.py](https://github.com/openmc-dev/openmc/blob/2f0a8941316bffb5c5a17ea525915ea5c2cb3ff9/openmc/plotter.py#L457-L458). So perhaps I have misunderstood as I assumed 'total' was MT 1

**Comment by paulromano** [comment]
Total is MT=1, and it is defined as MT=2 + MT=3. The link you have there for the ENDF manual is a bit out of date. If you look at a more recent version (https://doi.org/10.2172/1425114), you'll see in the sum rules table that 1 = 2 + 3, and 3 is defined as all that other stuff. The real error is that in plotter.py, when determining what reactions make up total, it doesn't go recursively through the sum rules to check the rule for MT=3. So, this is a bug, but not with the ENDF sum rules.

**Comment by eepeterson** [comment]
Thanks for posting this @Shimwell. I don't think the sum rules are to blame since if you access the cross sections directly from the hdf5 file you get the correct results (see below).
 `nucpath = openmc.data.DataLibrary.from_xml(cross_sections.xml).get_by_material('B10')['path']
inc_n = openmc.data.IncidentNeutron.from_hdf5(path)
xs_1 = inc_n[1].xs['294K'] 
xs_2 = inc_n[2].xs['294K'] 
xs_3 = inc_n[3].xs['294K'] 
`

I think this is an issue with the calculate_cexs function and associated logic and this function is only used in the `plotter.py` file so shouldn't have drastic ramifications

**Comment by eepeterson** [comment]
@paulromano beat me to the punch with the actual reason for this so thanks! @Shimwell are you planning to submit a fix?

**Comment by Shimwell** [comment]
Sorry I've not got around to a PR, not found a good way of solving this yet

**Comment by drewejohnson** [comment]
By similar logic, would this bug also apply when plotting redundant cross sections made up of redundant cross sections?

**Comment by eepeterson** [comment]
@drewejohnson I was just checking this and yes except fission seems to work fine MT=18, although absorption does need to be expanded and presumably capture as well

**Comment by drewejohnson** [comment]
Fission makes sense because none of it's components are redundant, just first through fourth chance fissions. But yeah, MT=27 and MT=101 look like they'll need similar treatment

**Comment by Shimwell** [comment]
> @drewejohnson I was just checking this and yes except fission seems to work fine MT=18, although absorption does need to be expanded and presumably capture as well

not concerned about

**Comment by Shimwell** [comment]
Sorry accidentally closed, but have now reopened.

Just looking at the test coverage this plotter.py is one of the lowest in the python coverage https://coveralls.io/builds/30664161/source?filename=openmc/plotter.py

**Comment by eepeterson** [comment]
Ok, but the logic in the calculate_cexs methods doesn't care about that nested redundancy. It simply checks to see if the MT value is in the library file reaction dictionary and there are some inconsistencies as to which redundant reactions are listed and which are not. For example, U235 explicitly has MT=3 in it's reaction list which means it isn't affected by this bug. However, B10 does not have MT=3. U235 also "has" MT=18 in it's reaction dict even though it is really comprised of 4 other reactions. Is there rhyme or reason for which redundant reactions are listed explicitly and which are not?

**Comment by Shimwell** [comment]
Often the MT numbers are just not in the original ace files and this differs from evaluation to evaluation.

There is also [ this the section of code ](https://github.com/openmc-dev/openmc/blob/2f0a8941316bffb5c5a17ea525915ea5c2cb3ff9/openmc/data/neutron.py#L460-L461) that decides on which MTs to keep or not when making the h5 files.

I tend to replace this part with range(1, 999) when generating nuclear data for [this nuclear cross section plotting website](xsplot.com) as I desire all the cross sections, but it would mean large h5 files are generated

**Comment by drewejohnson** [comment]
> Ok, but the logic in the calculate_cexs methods doesn't care about that nested redundancy

This is correct, and part of why I think we're seeing this issue. When we expand only one level deep when making the sum rules, we may have additional redundant reactions that the plotter is told should exist in the library

> The proper way to handle this I suppose would be to use the MT value if it is there and fall back to the sum rules if it's not

I would agree with this. It would certainly save error when building the plottable cross sections

**Comment by eepeterson** [comment]
With regards to the slowing-down power plotting option - it includes elastic and inelastic scattering, but I believe the derivation of xi being independent of energy and only dependent on the atomic weight ratio is only valid for elastic scattering. Somebody correct me if I'm wrong.

**Comment by nelsonag** [comment]
@eepeterson I originally put in that slowing-down-power option to investigate weaknesses in simply using the moderating ratio to compare a moderator to criticality limits, hence the incorporation of the inelastic channels.

You are correct that xi is defined for 0K, unbound, elastic scattering, and thus including an inelastic xs in xi * Sig_s would be incorrect. I used the elastic xi as a conservative (from a "which moderator is more likely to lead to criticality" perspective) estimation of a xi applicable to all scattering channels, the details of which probably dont matter here.

The correct way would be to include a xi that is derived using the outgoing energy distribution data, but that would probably be a significant amount of code that would never actually be used.

Having said that, either the slowing down power can be removed, or the inelastic component taken out, and I'd be ok with that.

By the way, thanks for identifying and tackling these changes @eepeterson, @Shimwell , @drewejohnson , and @paulromano! Also, that xsplot.com website is great; I'm going to be using that frequently!

**Comment by eepeterson** [comment]
@nelsonag thanks for the explanation. I just removed the inelastic part for the moment. I believe PR #1563 now plots correct cross-sections by checking to see if a redundant reaction is present in the library file and if not expanding using the sum rules. This way any redundant reaction that is explicitly included in the `IncidentNeutron.reactions` dictionary will be used rather than expanding the sum rules.

**Comment by Shimwell** [comment]
This issue has been fixed by #1563
